google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/AppEngineServerConfigurationType.java

```diff
@@ -72,12 +72,19 @@ public Icon getIcon() {
 
   @Override
   public ConfigurationFactory[] getConfigurationFactories() {
+    final ConfigurationFactory configurationFactory = super.getConfigurationFactories()[0];
+
     return new ConfigurationFactory[]{
         new ConfigurationFactory(this) {
+          @Override
+          public String getName() {
+            return configurationFactory.getName();
+          }
+
           @NotNull
           @Override
           public RunConfiguration createTemplateConfiguration(@NotNull Project project) {
-            return createJ2EEConfigurationTemplate(this, project, true /*isLocal*/);
+            return configurationFactory.createTemplateConfiguration(project);
```

Comment:
yes, ultimately it does the same thing. I think its better code to do the delegation like I updated it do instead of recreating it.